ui/hooks/useTransactionEventFragment.js

```diff
@@ -1,10 +1,28 @@
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
 
 import { useGasFeeContext } from '../contexts/gasFee';
-import { updateEventFragment } from '../store/actions';
+import {
+  createTransactionEventFragment,
+  updateEventFragment,
+} from '../store/actions';
+import { selectMatchingFragment } from '../selectors';
+import { TRANSACTION_EVENTS } from '../../shared/constants/transaction';
 
 export const useTransactionEventFragment = () => {
   const { transaction } = useGasFeeContext();
+  const fragment = useSelector((state) =>
+    selectMatchingFragment(state, {
+      fragmentOptions: {},
+      existingId: `transaction-added-${transaction?.id}`,
+    }),
+  );
+
+  useEffect(() => {
+    if (!fragment && transaction) {
+      createTransactionEventFragment(transaction.id, TRANSACTION_EVENTS.ADDED);
```

Comment:
Oops just noticed this should be .APPROVED -- .ADDED will end up firing the ADDED event which we don't want to capture when creating missing fragments. In case it was already fired.

Reply:
Just updated the PR as discussed.